app/models/podcast_planner.rb

```diff
@@ -0,0 +1,152 @@
+    @segment_count = params[:segment_count].try(:to_i)
+  end
+
+  def date_range_ends_by_episodes?
+    @date_range_condition == "episodes"
+  end
+
+  def date_range_ends_by_end_date?
+    @date_range_condition == "date"
+  end
+
+  def episodes_remain?
+    @dates.length < @number_of_episodes
+  end
+
+  def end_date_passed?(date)
+    date > @end_date
+  end
+
+  def periodic?
+    @week_condition == "periodic"
+  end
+
+  def monthly?
+    @week_condition == "monthly"
+  end
+
+  def ready_to_select_weeks?
+    if periodic?
+      @period.present?
+    elsif monthly?
+      @monthly_weeks.present?
+    else
+      false
+    end
+  end
+
+  def ready_to_select_date_range?
+    if date_range_ends_by_episodes?
+      @number_of_episodes.present?
+    elsif date_range_ends_by_end_date?
+      @end_date.present? && @end_date > @start_date
+    else
+      false
+    end
+  end
+
+  def ready_to_generate_dates?
+    @selected_days.present? &&
+      ready_to_select_weeks? &&
+      ready_to_select_date_range?
+  end
+
+  def generate_dates!
+    return unless ready_to_generate_dates?
+
+    if date_range_ends_by_episodes?
+      generate_dates_by_remaining_episodes
+    elsif date_range_ends_by_end_date?
+      generate_dates_by_end_date
+    end
+  end
+
+  def clear_dates!
+    @dates = []
+  end
+
+  def select_days_of_week_to_add(current_day)
+    dates_to_add = []
+    7.times do |day|
+      dates_to_add.push(current_day) if @selected_days.include?(current_day.wday)
+      current_day += 1.day
+    end
+    dates_to_add
+  end
+
+  def generate_dates_by_remaining_episodes
+    current_day = @start_date
+
+    while episodes_remain?
+      next_days = select_days_of_week_to_add(current_day)
+      if periodic?
+        @dates.concat(next_days)
+        current_day += @period.weeks
+      elsif monthly?
+        valid_next_days = next_days.select { |day| @monthly_weeks.include?(week_of_the_month(day)) }
+        @dates.concat(valid_next_days)
+        current_day += 1.week
+      end
+    end
+
+    @dates = @dates.slice(0, @number_of_episodes)
+  end
+
+  def generate_dates_by_end_date
+    current_day = @start_date
+
+    until end_date_passed?(current_day)
+      next_days = select_days_of_week_to_add(current_day)
+      if periodic?
+        @dates.concat(next_days)
+        current_day += @period.weeks
+      elsif monthly?
+        valid_next_days = next_days.select { |day| @monthly_weeks.include?(week_of_the_month(day)) }
+        @dates.concat(valid_next_days)
+        current_day += 1.week
+      end
+    end
+
+    @dates.select! { |date| date <= @end_date }
+  end
+
+  def week_of_the_month(date)
+    (date.day.to_f / 7).ceil
+  end
+
+  def ready_to_generate_drafts?
+    @dates.present? &&
+      @publish_time.present? &&
+      @segment_count.present?
+  end
+
+  def generate_drafts!
+    return unless ready_to_generate_dates?
+    return unless ready_to_generate_drafts?
+
+    @dates.each do |date|
```

Comment:
the logic here is that the differentiating aspect of these drafts is the date, so that's why i loop through the dates, and then just return general objects for the different attributes i'll want when actually making the db query in the controller to bulk insert.

Reply:
I'm kind of surprised `insert_all` doesn't let you pass in an array of `Episode.new` unsaved models.

Reply:
updated to use `Episode.new` here, and then the controller saves the models.